File: architeture-sample/basiclib/src/main/java/example/basiclib/util/ActivityUtils.java
package example.basiclib.util;

import android.app.Activity;
import android.content.Context;
import android.content.ContextWrapper;

/**
 * activity工具类
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */

public class ActivityUtils {

    public static Activity scanForActivity(Context context) {
        if (context == null)
            return null;
        else if (context instanceof Activity)
            return (Activity) context;
        else if (context instanceof ContextWrapper)
            return scanForActivity(((ContextWrapper) context).getBaseContext());

        return null;
    }
}

File: lib_base/src/main/java/com/effective/android/base/util/ActivityUtils.kt
package com.effective.android.base.util

import android.app.Activity
import android.content.Context
import android.content.ContextWrapper

/**
 * activity工具类
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */

object ActivityUtils {

    fun scanForActivity(context: Context?): Activity? {
        return when (context) {
            is Activity -> context
            is ContextWrapper -> scanForActivity(context)
            else -> null
        }
    }
}
